lib/tasks/spellcheck.rake

```diff
@@ -0,0 +1,93 @@
+#
+# Rake task for checking spelling in the documentation files.
+# By default checks all *.md and *.html files.
+#
+# Supports custom dictionaries:
+#
+#  - global dictionary located in the tasks gem (lib/tasks/.spell.dict)
+#  - repository specific dictionary (.spell.dict in the root directory)
+#
+# The custom dictionaries contains one word per line.
+# The lines starting with '#' character are ignored (used for comments),
+#
+
+CUSTOM_DICTIONARY_FILE = "spell.dict"
+
+# read the global and the repository custom dictionary
+def read_custom_words
+  # read the global default custom dictionary
+  dict_path = File.expand_path("../#{CUSTOM_DICTIONARY_FILE}", __FILE__)
+  custom_words = File.read(dict_path).split("\n")
+
+  if File.exist?(CUSTOM_DICTIONARY_FILE)
+    puts "Loading custom dictionary (#{CUSTOM_DICTIONARY_FILE})..." if verbose == true
+    # read the custom dictionary from the project directory
+    custom_words += File.read(CUSTOM_DICTIONARY_FILE).split("\n")
+  end
+
+  # remove comments
+  custom_words.reject! { |word| word.start_with?("#")}
+  # remove duplicates
+  custom_words.uniq!
+
+  custom_words
+end
+
+def aspell_speller
+  # raspell is an optional dependency, handle the missing case nicely
+  begin
+    require "raspell"
+  rescue LoadError
+    $stderr.puts "ERROR: Ruby gem \"raspell\" is not installed."
+    exit 1
+  end
+
+  # initialize aspell
+  speller = Aspell.new("en_US")
+  speller.suggestion_mode = Aspell::NORMAL
+  # ignore the HTML tags in the text
+  speller.set_option("mode", "html")
+
+  speller
+end
+
+namespace :check do
+  desc "Run spell checker (by default for *.md and *.html files in Git)"
```

Comment:
Um, POT file are little bit tricky - you need to generate it first, and it a has a specific format (gettext keywords and comments), we would have to preprocess the file.

Anyway, I'd rather add the spell check to rubocop-yast plugin (or even directly to upstream rubocop).